### ISSUE TITLE
The `no-redeclare` rule seems to not check the module scope.

### ISSUE THREAD

**Issue by mysticatea** [eslint/eslint#2903]
test.js

```js
export class A {}
export class A {}
```

.eslintrc

```json
{
    "ecmaFeatures": {"classes": true, "modules": true},
    "rules": {"no-redeclare": 2}
}
```

----

```
> eslint -v
v0.24.0
> eslint test.js --reset
```

## Actual

No errors.

## Expected

```
test.js
  2:13  error  A is already defined  no-redeclare

✖ 1 problem (1 error, 0 warnings)
```

----

Maybe, to fix, executes [this block](https://github.com/eslint/eslint/blob/master/lib/rules/no-redeclare.js#L47) when also the modules flag is turned on.

**Comment by nzakas** [comment]
Yeah, looks like a bug.

**Comment by mysticatea** [comment]
I will try to fix this.
